src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -332,9 +323,26 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
             out var requestId,
             out var traceState);
 
+        Activity? activity = _activitySource.CreateActivity(
+            ActivityName,
+            ActivityKind.Server,
+            string.IsNullOrEmpty(requestId) ? null! : requestId
+            );
+
+        if (activity is null)
+        {
+            if (loggingEnabled || diagnosticListenerActivityCreationEnabled)
+            {
+                activity = new Activity(ActivityName);
```

Comment:
I think the activity winds up unassociated with the requestId now because of the deletion at 337. We probably want this:
```C#
if(!string.IsNullOrEmpty(requestId)
{
    activity.SetParentId(requestId);
}
```